src/token/src/canister.rs

```diff
@@ -202,43 +202,47 @@ impl TokenCanister {
         self.state.borrow().stats.owner
     }
 
-    /// Returns an array of transaction records in range [start, start + limit) related to user `who`.
-    /// Unlike `getTransactions` function, the range [start, start + limit) for `getUserTransactions`
-    /// is not the global range of all transactions. The range [start, start + limit) here pertains to
+    /// Returns an array of transaction records in range [start, start + limit] related to user `who`.
+    /// Unlike `getTransactions` function, the range [start, start + limit] for `getUserTransactions`
+    /// is not the global range of all transactions. The range [start, start + limit] here pertains to
     /// the transactions of user who. Implementations are allowed to return less TxRecords than
     /// requested to fend off DoS attacks.
+    ///
+    /// # Arguments
+    /// * `who` - The user to get transactions for.
+    /// * `start` - The index of the first transaction to return.
+    /// * `limit` - The number of transactions to return.
     #[query]
     fn getUserTransactions(&self, who: Principal, start: Nat, limit: Nat) -> Vec<TxRecord> {
-        let mut transactions = vec![];
-
-        let limit_usize = limit.0.to_usize().unwrap_or(usize::MAX);
-        if limit_usize > MAX_TRANSACTION_QUERY_LEN {
-            ic_kit::ic::trap(&format!(
-                "Limit must be less then {}",
-                MAX_TRANSACTION_QUERY_LEN
-            ));
-        }
-
-        for tx in self.state.borrow().ledger.get_range(&start, &limit) {
-            if tx.from == who || tx.to == who || tx.caller == Some(who) {
-                transactions.push(tx.clone());
-            }
-        }
-
-        transactions
-    }
+        // If the start value is larger than usize then return an
+        // empty vec
+        let start = match start.0.to_usize() {
+            Some(start) => start,
+            None => return vec![],
+        };
+
+        // limit the query to `MAX_TRANSACTION_QUERY_LEN`
+        let limit = limit
+            .0
+            .to_usize()
+            .unwrap_or(MAX_TRANSACTION_QUERY_LEN)
+            .min(MAX_TRANSACTION_QUERY_LEN);
```

Comment:
this code is not needed.

Reply:
to convert , simply convert limit to u128 or something limit.into() or limit as u128.

Reply:
if start is that large we won't have any txs, i think it's ok to .expect("not that big!") or .into() and error

Reply:
Yeah , makes sense.